### PR TITLE
chore: support target directory

### PR DESCRIPTION
**Pull request by appleboy**
sync directory to custom target specific directory not only root path. I will create another PR to update the document.

### CODE REVIEW

**Comment by tboerger** [comment]
Can you explain the use case? For github pages it doesn't make sense to put it into another directory as the branch gets published.

**Comment by appleboy** [comment]
I want to create a homepage for the conference every year and the structure like as below:

gophercon.golangtw.org/ => The default home page and repo is `golangtw/gophercon`
gophercon.golangtw.org/2019 => repo: `golangtw/gophercon2019`
gophercon.golangtw.org/2020 => repo: `golangtw/gophercon2020`

Dron will auto-deploy `gophercon2019` to subfolder `2019` of `golangtw/gophercon` in `gh-pages` branch.

File: plugin.go [hunk 1]
```diff
@@ -135,14 +137,15 @@ func (p Plugin) cloneTarget() error {
...
 		"--delete",
 		"--exclude",
 		".git",
 		"--exclude",
 		"CNAME",
 		"-r",
-		p.Config.PagesDirectory,
-		p.Config.TemporaryBase,
+		strings.TrimRight(p.Config.PagesDirectory, "/")+"/",
+		path.Join(p.Config.WorkDirectory, p.Config.TargetDirectory),
```
Comment (tboerger) [inline plugin.go]: Did you mix something here? Before it used `p.Config.TemporaryBase` which is not used at all anymore?

Comment (appleboy) [reply plugin.go]: Just sync the all files in `p.Config.PagesDirectory + "/"` to target clone base not include the `p.Config.PagesDirectory`.

origin:

sync `p.Config.PagesDirectory` to `p.Config.TemporaryBase`

after:

sync `p.Config.PagesDirectory + "/"` to `p.Config.WorkDirectory`

Comment (tboerger) [reply plugin.go]: But the repo gets cloned to the temporary base so far?!

Comment (appleboy) [reply plugin.go]: Yes, the default behavior is sync all files in `PagesDirectory` include the self folder to `TemporaryBase`.

Change to sync all files in `PagesDirectory` **not** include the self folder. 

So you can define the `TargetFolder` to sync all files in `PagesDirectory` into `TargetFolder` folder base on `WorkDirectory`.

Comment (appleboy) [reply plugin.go]: example:

https://github.com/golangtw/gophercon2020/blob/7e37a6c300bbfca658e030544609b6fbbd319a67/.drone.yml#L23

```yaml
- name: publish
  image: appleboy/gh-pages
  pull: always
  settings:
    username:
      from_secret: username
    password:
      from_secret: password
    pages_directory: dist
    target_branch: master
    target_directory: 2020
    remote_url: https://github.com/golangtw/gophercon.github.io.git
  when:
    branch:
    - master
    event:
    - push
```

**Comment by donny-dont** [comment]
This should now be possible. Can you please confirm @appleboy

**Pull request closed without merging**